src/Generation/ResourcesGenerator.php

```diff
@@ -43,15 +44,18 @@ private static function ensureDecimal(string $s): string
 
     public static function generateDescriptorConfig(ServiceDetails $serviceDetails, GapicYamlConfig $gapicYamlConfig): string
     {
-        $perMethod = function ($method) use ($gapicYamlConfig, $serviceDetails) {
+        $preMigrationOnly = $serviceDetails->migrationMode == MigrationMode::PRE_MIGRATION_SURFACE_ONLY;
+        $perMethod = function ($method) use ($gapicYamlConfig, $serviceDetails, $preMigrationOnly) {
             switch ($method->methodType) {
                 case MethodDetails::CUSTOM_OP:
                     $descriptor = [
-                        'callType' => AST::literal('\Google\ApiCore\Call::LONGRUNNING_CALL'),
-                        // Include the responseType for Custom Operations because they define their own operation class.
-                        'responseType' => $method->responseType->getFullName(true),
                         'longRunning' => static::customOperationDescriptor($serviceDetails, $method)
                     ];
+                    if (!$preMigrationOnly) {
+                        // Include the responseType for Custom Operations because they define their own operation class.
+                        $descriptor['responseType'] = $method->responseType->getFullName(true);
+                        $descriptor['callType'] = AST::literal('\Google\ApiCore\Call::LONGRUNNING_CALL');
```

Comment:
I feel like you probably told me this but, is there any reason why "PRE_MIGRATION_SURFACE_ONLY" clients can't use these descriptors? They seem to be a V1 or V2 feature. Is it just because it's a change that will cause a lot of churn?

Reply:
Not all of the values are available in the older versions of GAX (requiring a GAX upgrade, complicating rollout of this stuff) and yes to avoid churn/extraneous config that isn't used by the client code at all.